src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -200,13 +202,67 @@ def __init__(self, experiment_name: str, device: str = None, multi_gpu: Union[Mu
         self.train_metrics, self.valid_metrics = default_train_metrics, default_valid_metrics
         self.loss_logging_items_names = default_loss_logging_items_names
 
-    def connect_dataset_interface(self, dataset_interface: DatasetInterface, data_loader_num_workers: int = 8):
+    def connect_dataset_interface(self, dataset_interface: DatasetInterface, data_loader_num_workers: int = 8,
+                                  fine_tune=False):
         """
         :param dataset_interface: DatasetInterface object
         :param data_loader_num_workers: The number of threads to initialize the Data Loaders with
             The dataset to be connected
+
+        :param fine_tune: This flag will ignore data_loader_num_workers.
+            When set to True, the program will search for a 'sweet spot' in terms of number of data loading processes.
+            The constellation that yields the highest data loading speed will be chosen.
+            This flag is useful when training multiple models over the same CPUs (same machine).
         """
         self.dataset_interface = dataset_interface
+        if fine_tune:
```

Comment:
Can we move extract this code block to a method in utils ? Does not fell like this logic should be inside connect_dataset_interface.

Reply:
Definitely, it was just here for easy context so you can understand it inline.
Will move ti another file after we agree on the logic.